docs/birthday-kata.md

```diff
@@ -0,0 +1,890 @@
+2) Sends an email to all employees whose birthday is today.
+
+Platform Goals:
+1) Testable; we should be able to test the internal application logic with no need to ever send a real email.
+2) Flexible: we anticipate that the data source in the future could change from a flat file to a relational database, or perhaps a web service. We also anticipate that the email service could soon be replaced with a service that sends greetings through Facebook or some other social network.
+3) Well-designed: separate clearly the business logic from the infrastructure.
+
+For further information about this kata please checkout the original post [here](http://matteo.vaccari.name/blog/archives/154).
+
+### Our Approach
+First, we will come up with a rich domain model and discuss different ways we can model the system to take advantage of the compiler nd what the tradeoffs are.
+
+Second, we will talk about Abilities, dependency injection / rejection, and how we can decouple possibly impure infrastructure from pure decision making logic.
+
+Third, we will apply TDD to implement our behavior.
+
+## Modeling
+Unison provides us with all the algebraic data types we need to build and compose types. In lieu of discussing them up front, we will just jump into modeling and discuss syntax and decisions as we go!
+
+From skimming over the prompt, it seems pretty clear that top level types we need are `Employee` and `Email`. Let's start with `Employee`.
+
+### Employee
+So what is an employee to us? For one, they are an entity and therefore should have a unique `id`. Some other fields we probably care about include the employee's `name`, and importantly, their `birthday`.
+
+Obviously, there a millions of other fields we could choose to include in our model, but we would also like to only include what we care about. In a larger more complex system, the Employee model may have many other fields we don't care about but make sense to include in the broader modelling context (see "bounded context", Domain Driven Design).
+
+So lets start with:
+```unison
+unique type Employee  = 
+  { id: EmployeeId
+  , name: Name
+  , birthday: Date 
+  }
+
+unique type EmployeeId = EmployeeId Text
+unique type FirstName = FirstName Text
+unique type LastName = LastName Text
+unique type Name = 
+  { first: FirstName
+  , last: LastName
+  }
+unique type Date = 
+  { day: Nat
+  , month: Nat
+  , year: Nat
+  }
+```
+#### Decision 1 - Record Types
+Our employee is a record type consisting of an `id`, `name`, and a `birthday`. In Unison, the record type syntax is actually equivalent to a normal data type like
+```unison
+type Employee = Employee EmployeeId Name Date
+```
+except the compiler will automatically generate a collection of functions (lens) that allow for easy access and updates of this immutable data structure, which will be a big help later. You can see this in the output in `ucm`:
```

Comment:
What if we omit the parenthetical about lenses here, in the spirit of "teach one thing" 😁?